### PR TITLE
Replace psutil.pid() with psutil.process_iter() for safer execution

### LINKED ISSUE #7514: Thread config tests not stable as it should
This exception shows up sometimes.
```
Traceback (most recent call last):
  File "/usr/local/lib64/python3.6/site-packages/psutil/_common.py", line 340, in wrapper
    ret = self._cache[fun]
AttributeError: _cache

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib64/python3.6/site-packages/psutil/_pslinux.py", line 1517, in wrapper
    return fun(self, *args, **kwargs)
  File "/usr/local/lib64/python3.6/site-packages/psutil/_common.py", line 343, in wrapper
    return fun(self)
  File "/usr/local/lib64/python3.6/site-packages/psutil/_pslinux.py", line 1559, in _parse_stat_file
    with open_binary("%s/%s/stat" % (self._procfs_path, self.pid)) as f:
  File "/usr/local/lib64/python3.6/site-packages/psutil/_common.py", line 604, in open_binary
    return open(fname, "rb", **kwargs)
FileNotFoundError: [Errno 2] No such file or directory: '/proc/19527/stat'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib64/python3.6/site-packages/psutil/__init__.py", line 474, in _init
    self.create_time()
  File "/usr/local/lib64/python3.6/site-packages/psutil/__init__.py", line 824, in create_time
    self._create_time = self._proc.create_time()
  File "/usr/local/lib64/python3.6/site-packages/psutil/_pslinux.py", line 1517, in wrapper
    return fun(self, *args, **kwargs)
  File "/usr/local/lib64/python3.6/site-packages/psutil/_pslinux.py", line 1729, in create_time
    ctime = float(self._parse_stat_file()['create_time'])
  File "/usr/local/lib64/python3.6/site-packages/psutil/_pslinux.py", line 1524, in wrapper
    raise NoSuchProcess(self.pid, self._name)
psutil.NoSuchProcess: psutil.NoSuchProcess process no longer exists (pid=19527)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "check_threads.py", line 126, in <module>
    main()
  File "check_threads.py", line 122, in main
    exit(count_threads(args.ts_path, args.etnet_threads, args.accept_threads, args.task_threads, args.aio_threads))
  File "check_threads.py", line 31, in count_threads
    p = psutil.Process(pid)
  File "/usr/local/lib64/python3.6/site-packages/psutil/__init__.py", line 447, in __init__
    self._init(pid)
  File "/usr/local/lib64/python3.6/site-packages/psutil/__init__.py", line 487, in _init
    raise NoSuchProcess(pid, None, msg)
psutil.NoSuchProcess: psutil.NoSuchProcess no process found with pid 19527
```

### PR DESCRIPTION
**Pull request by duke8253**
This fixes #7514.

### CODE REVIEW

File: tests/gold_tests/thread_config/check_threads.py [hunk 1]
```diff
@@ -23,12 +23,11 @@
 
 def count_threads(ts_path, etnet_threads, accept_threads, task_threads, aio_threads):
 
-    for pid in psutil.pids():
+    for p in psutil.process_iter():
```
Comment (bneradt) [inline tests/gold_tests/thread_config/check_threads.py:26]: Looks good.

As a tweak, reading the docs, it looks like the preferable way to do this is by passing in the subset of process attributes that are desired. In this case, perhaps:

```
for p in psutil.process_iter(['name', 'cwd', 'threads']):
   ...
   p.info['name'] == '[TS_MAIN]' and p.info['cwd'] == ts_path:
        ...
```

**Review by bneradt — CHANGES_REQUESTED** [review]
Looks good. Just a small suggestion. Thanks for addressing this @duke8253 !

File: tests/gold_tests/thread_config/check_threads.py (hunk 1 above)
Comment (duke8253) [reply tests/gold_tests/thread_config/check_threads.py]: done

Comment (bneradt) [reply tests/gold_tests/thread_config/check_threads.py]: Great. I'm glad you used p.name rather than p.info['name'] like I suggested. That's much nicer.

**Review by bneradt — APPROVED** [review]

**Review by randall — APPROVED** [review]

**Comment by bneradt** [comment]
### 9.0.x and 9.1.x cherry pick justification:

This is a simple test change that merges cleanly to 9.0.x and 9.1.x that makes the thread config tests more stable. I just triaged an autest-9.1.x failure that would have been fixed by this:

https://ci.trafficserver.apache.org/job/autest-9.1.x/9/console

Getting this into these branches will save people time figuring out why it failed.

**Pull request merged**